packages/web3torrent/src/clients/web3torrent-client.ts

```diff
@@ -102,23 +101,28 @@ export const parseMagnetURL: (rawMagnetURL: string) => Torrent = (rawMagnetURL =
   if (!rawMagnetURL.trim()) {
     return emptyTorrentData;
   }
-  const magnetURI = rawMagnetURL.trim().substring(1);
-  const magnetAsArray = magnetURI.split('&');
-
-  const magnetName = magnetAsArray.find(h => h.includes('dn=') || h.includes('name='));
-  const name = magnetName && magnetName.replace(/(dn=)|(name=)/g, '').replace(/\+/g, ' ');
+  const magnetParams = new URLSearchParams(rawMagnetURL.trim().replace(/#magnet:/g, ''));
 
-  const magnetLength = magnetAsArray.find(h => h.includes('xl='));
-  const length: number = Number(magnetLength && magnetLength.replace(/(xl=)/g, ''));
-
-  const magnetCost = magnetAsArray.find(h => h.includes('cost='));
-  const cost: number = Number(magnetCost && magnetCost.replace(/(cost=)/g, ''));
+  if (!magnetParams.has('tr')) {
+    defaultTrackers.map(tr => magnetParams.append('tr', tr));
+  }
 
   return {
     ...emptyTorrentData,
-    name,
-    magnetURI,
-    length: isNaN(length) ? 0 : length,
-    cost: isNaN(cost) ? 0 : cost
+    name: magnetParams.get('name') || magnetParams.get('dn') || 'unknown',
+    magnetURI: `magnet:?${magnetParams.toString()}`,
+    length: Number(magnetParams.get('xl')) || 0,
```

Comment:
Is it worth it to store this info as a number, considering that it's going to be part of a URL which is a string?

Reply:
I didn't give this much thought, 'cause I don't have clear how is this going to work. But... yeah, I could change this to be a string everywhere... I'll get to it.

Reply:
though length should be a number just the same. In the library the length is always the amount of bytes, and it's a number there, and I believe that it should be a number everywhere, URL notwithstanding